components/ToDos/ToDos.js

```diff
@@ -0,0 +1,27 @@
+import getAllTasks from "@/lib/data";
+import { uid } from "uid";
+import Task from "../Task/Task";
+import { useState } from "react";
+
+export default function ToDos() {
+  const defaultTasks = getAllTasks();
+  const sortedDefaultTasks = sortedByDate(defaultTasks);
+  function sortedByDate(array) {
+    const newArray = array.toSorted((a, b) => {
+      const dateA = new Date(a.dueDate);
+      const dateB = new Date(b.dueDate);
+      return dateA - dateB;
+    });
+    return newArray;
+  }
+  const [currentTasks, setCurrentTasks] = useState(sortedDefaultTasks);
+
+  return (
+    <main>
+      <h2>Tasks</h2>
+      {currentTasks.map((task) => (
+        <Task key={uid()} task={task} />
+      ))}
+    </main>
```

Comment:
I recommend to not include the `main` element inside of your component. You want to design your component as reusable as possible. The `main` element can be part of your page (in this case your homepage).
Additionally, you should not create an `id` here with `uid()` to use as a key. 
Using `uid()` inside the `map()` callback causes new IDs to be generated on every render, which is unnecessary and can cause issues with React’s reconciliation process.
Every task in your `data.js` should come already with an `id`, which can be used instead as a key here.

```suggestion
    <>
      <h2>Tasks</h2>
      {sortedDefaultTasks.map((task) => (
        <Task key={task.id} task={task} />
      ))}
    </>
```